File: client/src/lib/invoke/contactUpdate.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";

export function contactUpdate(
    login: string,
    nickname?: string | null,
    notes?: string | null,
    favorite?: boolean | null
): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("contact_update", { login, nickname, notes, favorite }),
        (error) => `Failed to update contact: ${error}`
    );
}

File: client/src-tauri/src/commands/contact_update.rs
use crate::contacts::ContactBook;

#[tauri::command]
pub fn contact_update(
    login: String,
    nickname: Option<String>,
    notes: Option<String>,
    favorite: Option<bool>,
) -> Result<(), String> {
    let mut book = ContactBook::load();
    book.update_contact(&login, nickname, notes, favorite)
}
